servers/parser-graphql-wrapper/src/databases/pocketMetadataStore.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  DynamoDBDocumentClient,
+  PutCommand,
+  QueryCommand,
+} from '@aws-sdk/lib-dynamodb';
+import config from '../config';
+import * as Sentry from '@sentry/node';
+import {
+  PocketMetadata,
+  PocketMetadataSource,
+} from '../__generated__/resolvers-types';
+import md5 from 'md5';
+
+export interface IPocketMetadataDataStore {
+  getStoredPocketMetadata(
+    resolvedUrl: string,
+    version: number,
+    source: PocketMetadataSource,
+  ): Promise<PocketMetadataEntity | null>;
+  storePocketMetadata(
+    input: PocketMetadataEntity,
+    ttl: number,
+  ): Promise<PocketMetadataEntity>;
+}
+
+export type PocketMetadataEntity = Omit<
+  PocketMetadata,
+  'createdAt' | 'datePublished'
+> & {
+  urlHash: string; // md5 hash of the resolved Url
+  createdAt: number; // epoch time in seconds
+  datePublished: number; // epoch time in seconds
+  version: number;
+  __typename: string;
+};
+
+export class ItemSummaryDataStoreBase implements IPocketMetadataDataStore {
+  public static table = config.dynamoDb.itemSummaryTable;
+
+  constructor(public conn: DynamoDBDocumentClient) {}
+  /**
+   * Fetch a share by ID from DynamoDB table
+   * @param id the lookup ID (hash key)
+   * @returns ShareEntity, or null if record does not exist
+   */
+  async getStoredPocketMetadata(
+    resolvedUrl: string,
+    version: number,
+    source: PocketMetadataSource,
+  ): Promise<PocketMetadataEntity | null> {
+    const getPreviewCommand = new QueryCommand({
+      TableName: ItemSummaryDataStoreBase.table.name,
+      KeyConditionExpression: `urlHash = :key`,
+      FilterExpression: `version = :version`,
+      ExpressionAttributeValues: {
+        ':key': md5(resolvedUrl),
+        ':version': version,
+      },
+    });
+    const response = await this.conn.send(getPreviewCommand);
+
+    const items: PocketMetadataEntity[] = response.Items.filter(
+      (item: PocketMetadataEntity) => {
+        // source is a reserved keyword in dynamo and we want to project all values,
+        // so we just filter after instead of filtering at the dynamodb level
+        return item.source == source;
```

Comment:
I don't think you're ever going to have more than one record... unless I'm mistaking how this table is keyed

Reply:
Yea that was mainly the thinking, just in case we change one url to use a different parser